src/logic/useMobility.ts

```diff
@@ -0,0 +1,77 @@
+import { useState, useEffect, useMemo } from 'react'
+import { Coordinates } from '@entur/sdk'
+
+import { Vehicle, FormFactor, Operator } from '@entur/sdk/lib/mobility/types'
+
+import service from '../service'
+import { useSettingsContext } from '../settings'
+import { REFRESH_INTERVAL, ALL_ACTIVE_OPERATOR_IDS } from '../constants'
+
+import { useOperators } from '.'
+
+async function fetchVehicles(
+    coordinates: Coordinates,
+    distance: number,
+    operators: Operator[],
+    formFactors: FormFactor[] | undefined,
+): Promise<Vehicle[]> {
+    if (!coordinates || !distance || !operators?.length) {
+        return []
+    }
+
+    const vehicles: Promise<Vehicle[]> = service.mobility.getVehicles({
+        lat: Number(coordinates.latitude),
+        lon: Number(coordinates.longitude),
+        range: distance,
+        count: 50,
+        operators: operators.map((operator) => operator.id),
+        formFactors: formFactors ? formFactors : undefined,
+    })
+
+    return vehicles
+}
+
+export default function useMobility(
+    formFactors: FormFactor[] | undefined = undefined,
+): Vehicle[] | null {
+    const [settings] = useSettingsContext()
+    const allOperators = useOperators(Object.values(ALL_ACTIVE_OPERATOR_IDS))
+    const [vehicles, setVehicles] = useState<Vehicle[] | null>([])
+
+    const { coordinates, distance, hiddenMobilityOperators, hiddenModes } =
+        settings || {}
+
+    const operators = useMemo(
+        () =>
+            allOperators.filter(
+                (operator) =>
+                    !hiddenMobilityOperators ||
+                    !hiddenMobilityOperators?.includes(operator.id),
+            ),
+        [hiddenMobilityOperators, allOperators],
+    )
+
+    const isDisabled = Boolean(hiddenModes?.includes('sparkesykkel'))
+
+    useEffect(() => {
+        if (!coordinates || !distance || isDisabled) {
+            return setVehicles(null)
+        }
+
+        fetchVehicles(coordinates, distance, operators, formFactors)
+            .then(setVehicles)
+            // eslint-disable-next-line no-console
+            .catch((error) => console.error(error))
+
+        const intervalId = setInterval(() => {
+            fetchVehicles(coordinates, distance, operators, formFactors).then(
+                setVehicles,
+            )
+        }, REFRESH_INTERVAL)
+
+        return (): void => clearInterval(intervalId)
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
1: ignoring suggestion to include prop in dep-array

Reply:
Me bør helst unngå å ignorere denne feilen. Eg antar det er `formFactors` den klagar på. Grunnen til at det vert evig løkke er sannsynligvis fordi `useMobility` blir alltid kalt med ein heilt ny array-referanse: `useMobility([FormFactor.SCOOTER])`, som så vil alltid trigge ny useEffect-kjøring, sjølv om innholdet i arrayet er likt. 

Det er mange måter å løyse det på, f.eks. ved å dra ut `[FormFactor.SCOOTER]` til ein `const` der du bruker hooken, eller det enklaste her er kanskje å skrive om argumentet til å kun ta inn 1 formFactor. Det ser ikkje ut som at `useMobility` blir kalt med fleire enn 1 så langt, i alle fall.